modules/lightmapper_rd/lm_compute.glsl

```diff
@@ -347,7 +349,7 @@ float get_omni_attenuation(float distance, float inv_range, float decay) {
 	return nd * pow(max(distance, 0.0001), -decay);
 }
 
-void trace_direct_light(vec3 p_position, vec3 p_normal, uint p_light_index, bool p_soft_shadowing, out vec3 r_light, out vec3 r_light_dir, inout uint r_noise) {
+void trace_direct_light(vec3 p_position, vec3 p_normal, uint p_light_index, bool p_soft_shadowing, out vec3 r_light, out vec3 r_light_dir, out float r_shadow, inout uint r_noise) {
```

Comment:
`r_shadow` is an `out` parameter, but the function can return early without writing to it, in which case its value will be undefined when used later in the calling function. It should have a valid value in all code paths, for example `r_light` initializes its default value at the start of the function to make sure it always has a valid value.